book/src/user_manual/group_config.md

```diff
@@ -33,3 +34,7 @@ Example create configuration:
 ```rust,no_run,noplayground
 {{#include ../../../openmls/tests/book_code.rs:mls_group_create_config_example}}
 ```
+
+## Unknown extensions
+
+Some extensions carry data, but don't alter the behaviour of the protocol (e.g.  the application_id extension). OpenMLS allows the use of arbitrary such extensions in the group context, key packages and leaf nodes. Such extensions can be instantiated and retrieved through the use of the `UnknownExtension` struct and the `ExtensionType::Unknown` extension type. Such "unknown" extensions are handled transparently by OpenMLS, but can be used by the application, e.g. to have a group agree on pieces of data.
```

Comment:
Supporting arbitrary extensions in the GroupContext is not RFC compliant. Extensions in the GroupContext need to be supported by every member (ie, also have an indication of support in their `capabilities` field).

Reply:
Why do you think that's not RFC compliant?
Indicating support in capabilities if of course required. But MLS does not restrict extensions. In the contrary, implementations are supposed to handle any (unknown) extension gracefully. That's what the grease mechanisms tries to ensure.

Reply:
I am not sure what you mean; the two statements do not seem to contradict each other. OpenMLS can just carry the extension data payload as a bunch of bytes, as long as the extension type is part of the required capabilities in the group context extensions. If it is not in the required capabilities, a proposal to add the unknown extension to the group context extension list is considered invalid by OpenMLS.

Reply:
It says in section 13.4:
- Clients must verify they support every extension in a GroupContext. Otherwise, it must refuse to join a group
- Clients must verify every other client supports extensions in the GroupContext

Reply:
Is the documentation a mis-statement, or does openmls actually allow arbitrary GroupContext extensions?

Reply:
It allows arbitrary group context extension, as long as all members support them.
As such, it is compliant with the RFC. The difference to other extensions is that the MLS implementation isn't aware of them. But it doesn't need to be as long as they are only relevant for the application. This way using especially private extension (0xF000 - 0xFFFF) becomes much easier, because the protocol implementation doesn't need to be made aware of them.
That said, @keks found an issue in the validation check for this (see https://github.com/openmls/openmls/issues/1487). This is part of the larger effort of getting all validation checks into OpeMLS (see the [validation label](https://github.com/openmls/openmls/labels/validation) for some of them).